### PR TITLE
UnusedFlowRule

### PR DESCRIPTION
**Pull request by thedevnisarg**

### CODE REVIEW

File: src/main/groovy/com/avioconsulting/mule/linter/model/Application.groovy [hunk 1]
```diff
@@ -61,6 +62,42 @@ class Application {
         }
     }
 
+    List<MuleComponent> findComponents(String componentType, String namespace) {
+        List<MuleComponent> comps = []
+        configurationFiles.each { configFile ->
+            comps += configFile.findComponents(componentType, namespace)
+        }
+        return comps
```
Comment (adesjardin) [inline src/main/groovy/com/avioconsulting/mule/linter/model/Application.groovy:70]: Would the .collect() and flatten() methods work here?  `return configurationFiles.collect { configFile.findComponents(componentType, namespace) }.flatten()`

Comment (adesjardin) [reply src/main/groovy/com/avioconsulting/mule/linter/model/Application.groovy:70]: This pattern applies to the getFlows, getSubFlows, and getFlowrefs as well.

File: src/main/groovy/com/avioconsulting/mule/linter/model/Application.groovy [hunk 2]
```diff
@@ -61,6 +62,42 @@ class Application {
...
+    }
+
+    List<FlowComponent> getFlows() {
+        List<FlowComponent> comps = []
+        configurationFiles.each { configFile ->
```
Comment (adesjardin) [inline src/main/groovy/com/avioconsulting/mule/linter/model/Application.groovy:75]: `return configurationFiles.collect { it.flows }.flatten() `

File: src/main/groovy/com/avioconsulting/mule/linter/model/Application.groovy (hunk 1 above)
Comment (thedevnisarg) [reply src/main/groovy/com/avioconsulting/mule/linter/model/Application.groovy]: Yap it does. WOW! much more simplified.

File: src/main/groovy/com/avioconsulting/mule/linter/model/Application.groovy (hunk 2 above)
Comment (thedevnisarg) [reply src/main/groovy/com/avioconsulting/mule/linter/model/Application.groovy]: done.

**Pull request merged**